src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -176,7 +183,9 @@ module.exports = function<T, P, I, TI, C>(config : HostConfig<T, P, I, TI, C>, s
     }
     // Rerender
     const instance = workInProgress.stateNode;
+    ReactCurrentOwner.current = workInProgress;
     const nextChildren = instance.render();
+    ReactCurrentOwner.current = null;
```

Comment:
This is doesn't reset it if the render throws. We should probably just move `ReactCurrentOwner.current = null;` out to @gaearon's top level try-block in #8095. So that we just reset it when we're completely done reconciling.